martha/martha_v3.js

```diff
@@ -6,156 +6,207 @@ const url = require('url');
-            updated_time: response.data_object.updated,
-        };
-    } else {
-        throw new FailureResponse(400, `Expected but did not receive properly formatted DOS Response: ${JSON.stringify(response)} `);
-    }
-}
+function responseParser (response) {
+    // If this is not a DOS response, assume it's already DRS and return it.
+    if (!response.data_object) { return response; }
 
-function drsResponseParser (response) {
-    return {
-        access_methods: response.access_methods,
-        checksums: response.checksums,
-        created_time: response.created_time,
-        mime_type: response.mime_type,
-        name: response.name,
-        size: Number(response.size),
-        updated_time: response.updated_time,
-    };
+    // Otherwise, find the DOS fields and convert them to DRS.
+    const {
+        urls,
+        checksums,
+        created: created_time,
+        mimeType: mime_type,
+        name,
+        size,
+        updated: updated_time,
+    } = response.data_object;
+    const access_methods =
+        urls &&
+        urls
+            .filter((e) => e.url.startsWith('gs://'))
+            .map((gsUrl) => { return { type: 'gs', access_url: { url: gsUrl.url } }; });
+    return { access_methods, checksums, created_time, mime_type, name, size, updated_time };
 }
 
 /** *************************************************************************************************
  * Here is where all the logic lives that pairs a particular kind of URI with its
  * resolving-URL-generating parser, what path to use to make a Bond request for an SA (if any), and
  * a response parser.
+ *
+ * If you update this function update the README too!
  */
 
 function determineDrsType (parsedUrl) {
-    if ((parsedUrl.host.toLowerCase() === 'dg.4503') || (parsedUrl.host.toLowerCase() === 'dg.712c')) {
+    const host = parsedUrl.hostname.toLowerCase();
+
+    // First handle servers that we know about...
+
+    // Compact BDC
+    if (['dg.4503', 'dg.712c'].includes(host)) {
         return new DrsType(
-            dosCompactUrlGenerator(parsedUrl),
-            false,
-            `${config.bondBaseUrl}/api/link/v1/fence/serviceaccount/key`,
-            dosResponseParser);
-    } else if (parsedUrl.host.toLowerCase().startsWith('dg.')) {
+            DG_EXPANSION_BIO_DATA_CATALYST,
+            PROTOCOL_PREFIX_DOS,
+            AUTH_SKIPPED,
+            BOND_PROVIDER_FENCE,
+        );
+    }
+
+    // Full BDC
+    if (host === DG_EXPANSION_BIO_DATA_CATALYST) {
         return new DrsType(
-            dosCompactUrlGenerator(parsedUrl),
-            false,
-            `${config.bondBaseUrl}/api/link/v1/dcf-fence/serviceaccount/key`,
-            dosResponseParser);
-    } else if (parsedUrl.host.endsWith('.dataguids.org')) {
+            DG_EXPANSION_NONE,
+            PROTOCOL_PREFIX_DOS,
+            AUTH_SKIPPED,
+            BOND_PROVIDER_FENCE,
+        );
+    }
+
+    // Compact the AnVIL
```

Comment:
nit: why the "the"?

Reply:
🤷 Go ask the AnVIL: https://gen3.theanvil.io/